WooCommerce/Classes/Blaze/BlazeCreationEntryCoordinator.swift

```diff
@@ -0,0 +1,173 @@
+            } else if fetchedObjects.count == 1, let firstProduct = fetchedObjects.first {
+                return .campaignForm(productID: firstProduct.productID)
+            }
+            else {
+                return .noProductAvailable
+            }
+        }
+    }
+
+    /// Handles navigation to the native Blaze creation
+    func navigateToNativeCampaignCreation(source: BlazeSource, productID: Int64) {
+        let viewModel = BlazeCampaignCreationFormViewModel(siteID: self.siteID,
+                                                           productID: productID,
+                                                           onCompletion: self.onCampaignCreated)
+        let controller = BlazeCampaignCreationFormHostingController(viewModel: viewModel)
+
+        // This function can be called from navigateToBlazeProductSelector(), in which case we need to show the
+        // Campaign Creation Form from blazeNavigationController.
+        // Otherwise, we show it from the current navigation controller.
+        if blazeNavigationController.presentingViewController != nil {
+            blazeNavigationController.show(controller, sender: self)
+        } else {
+            navigationController.show(controller, sender: self)
+        }
+    }
+
+    /// Handles navigation to the webview Blaze creation
+    func navigateToWebCampaignCreation(source: BlazeSource, productID: Int64?) {
+        let webViewModel = BlazeWebViewModel(siteID: siteID,
+                                             source: source,
+                                             siteURL: siteURL,
+                                             productID: productID) { [weak self] in
+            guard let self else { return }
+            self.onCampaignCreated()
+        }
+        let webViewController = AuthenticatedWebViewController(viewModel: webViewModel)
+        navigationController.show(webViewController, sender: self)
+        didSelectCreateCampaign?(source)
+    }
+
+    /// Handles navigation to the Blaze product selector view
+    func navigateToBlazeProductSelector(source: BlazeSource) {
+        let controller: ProductSelectorViewController = {
+            let productSelectorViewModel = ProductSelectorViewModel(
+                siteID: siteID,
+                onProductSelectionStateChanged: { [weak self] product in
+                    guard let self = self else { return }
+
+                    // Navigate to Campaign Creation Form once any type of product is selected.
+                    navigateToNativeCampaignCreation(source: source, productID: product.productID)
+                },
+                onCloseButtonTapped: { [weak self] in
+                    guard let self = self else { return }
+
+                    navigationController.dismiss(animated: true, completion: nil)
+                }
+            )
+            return ProductSelectorViewController(configuration: ProductSelectorView.Configuration.configurationForBlaze,
```

Comment:
Nit: this can be simplified

```suggestion
            return ProductSelectorViewController(configuration: .configurationForBlaze,
```